# Simple ATLAS Analysis

Crude form to fetch ATLAS data and plot it

In [1]:
from func_adl_servicex import ServiceXSourceUpROOT
from func_adl import ObjectStream
from servicex import ignore_cache
from hist import Hist
import mplhep as mpl
import awkward as ak

from utils import files

In [2]:
def make_ds(name: str):
    '''Create a ServiceX Datasource for a particular ATLAS Open data file
    '''
    return ServiceXSourceUpROOT(files[name]['files'], files[name]['treename'], backend_name='uproot-af')

In [3]:
def apply_event_cuts (source: ObjectStream) -> ObjectStream:
    '''Event level cuts for the analysis. Keep from sending data that we aren't going to need at all in the end.
    '''
    return (source
        .Where(lambda e: e.trigE or e.trigM))

In [4]:
ds = make_ds('ggH125_ZZ4lep')
good_events = apply_event_cuts(ds)

In [5]:
data = (good_events
    .Select(lambda e: {
        'lep_pt': e['lep_pt']
        })
    .AsAwkwardArray()
    .value())

In [6]:
len(data['lep_pt']), len(ak.flatten(data['lep_pt']))

(164716, 659806)

Some info on how the cuts are going

| cuts | # Events | # Leptons |
| -: | -: | -: |
| None | 164,716 | 659806 |
| E/M Trigger |  164,715 | 659802 |

Total datasize of the full query: 25 MB


In [7]:
ak.count(data['lep_pt'], axis=1)

<Array [4, 4, 4, 4, 4, 4, ... 4, 4, 4, 4, 4, 4] type='164716 * int64'>